BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -851,15 +898,17 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
 
             return identityTokenResponse
         } catch let error as IdentityTokenRequestError {
-            if case let .twoFactorRequired(_, ssoToken, captchaBypassToken) = error {
+            if case let .twoFactorRequired(_, captchaBypassToken, masterPasswordPolicyResponseModel, ssoToken) = error {
                 // If the token request require two-factor authentication, cache the request so that
                 // the token information can be added once the user inputs the code.
                 twoFactorRequest = request
                 twoFactorRequest?.captchaToken = captchaBypassToken
 
                 // Form the resend email request in case the user needs to resend the verification code email.
                 var passwordHash: String?
-                if case let .password(_, password) = request?.authenticationMethod { passwordHash = password }
+                if case let .password(_, password) = request?.authenticationMethod { passwordHash = password
+                    try await checkMasterPasswordPolicies(masterPassword!, true, email, masterPasswordPolicyResponseModel)
```

Comment:
⚠️ We try never to use `!` as it can be dangerous if, for some reason `masterPassword` doesn't have a value, so add an `if let` check or add a `guard let` check which logs the error if that happens depending on whether you consider an error should be logged if `masterPassword` is `nil` and provide some description on such error so if we encounter that in Crashlytics we understand the context.